modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaSearchVocabulariesMethodCheck.java

```diff
@@ -31,64 +35,72 @@ protected String doProcess(
 
 		String newContent = content;
 
-		boolean failedValidation = false;
-		boolean replaced = false;
+		JavaClass javaClass = JavaClassParser.parseJavaClass(fileName, content);
 
-		Matcher matcher = _searchVocabulariesPattern.matcher(content);
+		for (JavaTerm childJavaTerm : javaClass.getChildJavaTerms()) {
+			if (!childJavaTerm.isJavaMethod()) {
+				continue;
+			}
 
-		while (matcher.find()) {
-			String methodCall = JavaSourceUtil.getMethodCall(
-				content, matcher.start());
+			JavaMethod javaMethod = (JavaMethod)childJavaTerm;
 
-			if (!hasClassOrVariableName(
-					"AssetVocabularyService", content, content, methodCall) &&
-				!hasClassOrVariableName(
-					"AssetVocabularyLocalService", content, content,
-					methodCall)) {
+			String javaMethodContent = javaMethod.getContent();
 
-				continue;
-			}
+			Matcher matcher = _searchVocabulariesPattern.matcher(
+				javaMethodContent);
 
-			List<String> parameterList = JavaSourceUtil.getParameterList(
-				methodCall);
+			while (matcher.find()) {
+				String methodCall = JavaSourceUtil.getMethodCall(
+					javaMethodContent, matcher.start());
 
-			if (parameterList.size() != 5) {
-				continue;
-			}
+				if (!hasClassOrVariableName(
+						"AssetVocabularyService", content, content,
+						methodCall) &&
+					!hasClassOrVariableName(
+						"AssetVocabularyLocalService", content, content,
+						methodCall)) {
 
-			String[] parameterTypes = {"long", "long", "String", "int", "int"};
+					continue;
+				}
 
-			if (!hasParameterTypes(
-					content, content, ArrayUtil.toStringArray(parameterList),
-					parameterTypes)) {
+				List<String> parameterList = JavaSourceUtil.getParameterList(
+					methodCall);
 
-				failedValidation = true;
+				if (parameterList.size() != 5) {
+					continue;
+				}
 
-				continue;
-			}
+				String[] parameterTypes = {
+					"long", "long", "String", "int", "int"
+				};
 
-			String newMethod = _addOrReplaceParameters(
-				matcher.group(0), methodCall, parameterList);
+				if (!hasParameterTypes(
+						javaMethodContent, javaMethodContent,
+						ArrayUtil.toStringArray(parameterList),
+						parameterTypes)) {
 
-			newContent = StringUtil.replace(newContent, methodCall, newMethod);
+					addMessage(
+						fileName,
+						StringBundler.concat(
+							"Could not resolve types of searchVocabularies ",
+							"method. The method signature has changed to ",
+							"searchVocabularies(long companyId, long[] ",
+							"groupIds, String title, int[] visibilityTypes, ",
+							"int start, int end, Sort sort). Fill the new ",
+							"parameters manually."));
```

Comment:
@TamyrisTorres, can we rephrase this message as was done in https://github.com/natocesarrego/liferay-portal/pull/1343#discussion_r1290079410?